src/Core/DecimalFloatComparison.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <base/DecomposedFloat.h>
+
+namespace DB
+{
+struct DecimalFloatComparison
+{
+    template <typename Float, typename Int>
+    static int compare(Float a, Int b, Int scale)
+    {
+        /// TODO need to implement comparison
+        if (a)
+            return -1;
+        if (b)
+            return 0;
+        if (scale)
+            return 1;
+        return 0;
+    }
```

Comment:
Now, I left this function to implement, could you first take a look, does current design is ok? @CurtizJ

Reply:
However, I still have no clear idea about the implementation, I checked some systems, in MySQL and Spark, they always convert decimal into float before comparing.